src/arch/util/machine-common-core.C

```diff
@@ -1505,15 +1505,9 @@ static void ConverseRunPE(int everReturn) {
  * The reason to initialize this variable here:
  * cmiArgDebugFlag is possibly accessed in CmiPrintf/CmiError etc.,
  * therefore, we have to initialize this variable before any calls
- * to those functions (such as CmiPrintf). Otherwise, we may encounter
- * a memory segmentation fault (bad memory access). Note, even
- * testing CpvInitialized(cmiArgDebugFlag) doesn't help to solve
- * this problem because the variable indicating whether cmiArgDebugFlag is
- * initialized or not is not initialized, thus possibly causing another
- * bad memory access. --Chao Mei
+ * to those functions (such as CmiPrintf).
  */
-  CpvInitialize(int, cmiArgDebugFlag);
-  CpvAccess(cmiArgDebugFlag) = 0;
+  cmiArgDebugFlag = 0;
```

Comment:
This is a race condition as each pe sets it to 0 and then possibly back to 1. In theory it could have an inconsistent value in the meantime.

Reply:
Since `cmiArgDebugFlag` is now a global variable, there is no need to assign it to zero here.